engine/execution/computation/computer/computer_test.go

```diff
@@ -344,6 +345,67 @@ func TestBlockExecutor_ExecuteBlock(t *testing.T) {
 		require.NoError(t, err)
 		assert.Len(t, result.StateSnapshots, collectionCount+1) // +1 system chunk
 	})
+	t.Run("forks are not sharing programs", func(t *testing.T) {
+
+		logger := zerolog.Nop()
+
+		ctx := fvm.NewContext(
+			logger,
+			fvm.WithTransactionProcessors(
+				fvm.NewTransactionInvocator(logger),
+			),
+		)
+
+		contractLocation := common.AddressLocation{
+			Address: common.Address{0x1},
+			Name:    "Test",
+		}
+
+		// TODO fix me to be distinct
+		contractProgram1 := &interpreter.Program{Program: ast.NewProgram(nil)}
+		contractProgram2 := &interpreter.Program{}
+
+		const transactionCount = 1
+		const collectionCount = 1
+		var executionCalls int
+
+		rt := &testRuntime{
+			executeTransaction: func(script runtime.Script, r runtime.Context) error {
+				executionCalls++
+				if executionCalls == 0 {
+					err := r.Interface.SetProgram(
+						contractLocation,
+						contractProgram1,
+					)
+					require.NoError(t, err)
```

Comment:
Probably want to return after this, or is it intentional that the second `SetProgram` is called too?

Reply:
I removed this test since it was not related to this pr, I added a bigger set of tests to Maks's PR